packages/trie/src/baseTrie.ts

```diff
@@ -566,11 +572,13 @@ export class Trie {
       // is applied twice (performance)
       const hashRoot = keccak(rlpNode)
 
-      if (remove && this.isCheckpoint) {
-        opStack.push({
-          type: 'del',
-          key: hashRoot,
-        })
+      if (remove) {
+        if (this._deleteFromDB) {
+          opStack.push({
+            type: 'del',
+            key: hashRoot,
+          })
+        }
```

Comment:
Note that this change is also changing the semantics of this conditional clause. Before if - when using a `CheckpointTrie` - being not within a checkpoint the `remove` calls would have been transitioned into a `put` operation. I am not seeing the logic of this and would assume this is falsy behaviour with no side effects (or maybe with side effects, not 100% sure), since the node references are removed from the trie anyhow (so it has no effect if the value is saved (again) in the DB. I wouldn't exclude the possibility though that I am overlooking something here.

Reply:
that is a good observation and indeed weird